### PR TITLE
disable SVM on Nvidia CL when running pytest

### PR DESCRIPTION
**Pull request by matthiasdiener**
Partially addresses #196.

### CODE REVIEW

**Comment by inducer** [comment]
LGTM, thanks!

File: arraycontext/impl/pytato/__init__.py [hunk 1]
```diff
@@ -260,26 +260,37 @@ def __init__(
...
 
         if allocator is None:
+            import sys
+
             from pyopencl.characterize import has_coarse_grain_buffer_svm
             has_svm = has_coarse_grain_buffer_svm(queue.device)
-            if has_svm:
-                self.using_svm = True
+            is_nvidia_cl_with_pytest = (queue.device.platform.name == "NVIDIA CUDA"
+                                        and "pytest" in sys.modules)
```
Comment (alexfikl) [inline arraycontext/impl/pytato/__init__.py:268]: Shouldn't this logic be in one of the `PytestArrayContextFactory`? It feels weird to put it in here.

Comment (inducer) [reply arraycontext/impl/pytato/__init__.py:268]: I agree that that's probably better.

Comment (matthiasdiener) [reply arraycontext/impl/pytato/__init__.py]: What do you think of https://github.com/inducer/arraycontext/pull/200/commits/618f16a20c8aad0a8a89f3a306f8aba05fb6d8ee ?

Comment (alexfikl) [reply arraycontext/impl/pytato/__init__.py]: Much nicer!

**Pull request merged**